server/embed.go

```diff
@@ -2,17 +2,35 @@ package server
 
 import (
 	"fmt"
-	"github.com/bakape/meguca/common"
-	"net/http"
 	"strings"
+	"database/sql"
+	"net/http"
+
+	"github.com/bakape/meguca/common"
+	"github.com/bakape/meguca/db"
 
 	"github.com/badoux/goscraper"
-	"github.com/otium/ytdl"
+	"github.com/xenking/ytdl"
 )
 
 // Get YouTube video information by ID
 func youTubeData(w http.ResponseWriter, r *http.Request) {
-	info, err := getYouTubeInfo(extractParam(r, "id"))
+	var info string
+	id := extractParam(r, "id")
+	err := db.InTransaction(false, func(tx *sql.Tx) error {
+		title, thumb, video, videoHigh, err := db.GetYouTubeInfo(tx, id)
+
+		if err != nil {
+			title, thumb, video, videoHigh, err = getYouTubeInfo(tx, id)
```

Comment:
Why are you linking the lifetime of external API requests and a DB transaction? This can increase lock row contention and connection pool utilization needlessly.  

Please do a cheap row existence check in its own transaction and then fetch data, if there is none. That data should be written in one query.